pyadjoint/reduced_functional.py

```diff
@@ -16,23 +46,35 @@ class ReducedFunctional(object):
             usually :class:`AdjFloat`. This should be the return value of the
             functional you want to reduce.
         controls (list[Control]): A list of Control instances, which you want
-            to map to the functional. It is also possible to supply a single Control
-            instance instead of a list.
-
+            to map to the functional. It is also possible to supply a single
+            Control instance instead of a list.
+        derivative_components (tuple of int): The indices of the controls with
+            respect to which to take the derivative. By default, the derivative
+            is taken with respect to all controls. If present, it overwrites
+            derivative_cb_pre and derivative_cb_post.
+        derivative_cb_pre (function): Callback function before evaluating
+            derivatives. Should return a list of Controls (usually the same
+            list as the input) to be passed to compute_gradient.
+        derivative_cb_post (function): Callback function after evaluating
+            derivatives. Should return a list of Controls (usually the same
+            list as the input) to be passed to copmute_gradient.
```

Comment:
Is this really the interface for the post callback?

Reply:
What arguments to the callbacks take?

Reply:
fixed in latest commit.